Add optional paramaters to functions

nph4rd opened the pull request:
There are some functions that do not have the optional parameters enabled. 

This PR will add them. 

Closes: #8

nph4rd committed:
Add aggregate param to order_book

nph4rd committed:
Fix order book example

nph4rd committed:
Add optional params to get_trades

nph4rd committed:
Add comment about uncovered sections

nph4rd committed:
Run fmt

nph4rd committed:
Add optional params to get_ledger

nph4rd committed:
Add optional parameters to get_withdrawals method

nph4rd committed:
Use OptionalParams struct to avoid too many args in methods

nph4rd committed:
Add optional parameters to get_fundings method

nph4rd committed:
Add optional parameters to get_user_trades method

nph4rd committed:
Add optional parameters for get_order_trades method

nph4rd committed:
Add optional parameters to get_open_orders method

nph4rd committed:
Add optional parameters to get_lookup_orders method

nph4rd committed:
Add optional parameters to cancel_order method

nph4rd committed:
Add optional parameters to place_order method

nph4rd committed:
Add OptionalOrderParams struct

This avoids having a "too many parameters" warning
from Clippt in the place_order method.

mucinoab commented on src/client.rs:
```diff
@@ -55,6 +55,24 @@ pub enum ApiError {
+}
+
+/// Optional parameters for an order.
+/// For more info see: <https://bitso.com/api_info?python#place-an-order>
```
I think we could remove the "python" from the url. 
"https://bitso.com/api_info?place-an-order#place-an-order"

nph4rd replied:
Ah! Thank you!

mucinoab commented on src/client.rs:
```diff
@@ -486,20 +648,32 @@ impl Bitso {
-        let result = self
-            .get(&url, &mut HashMap::new(), ApiType::Private)
-            .await?;
+        println!("{:?}", url);
```
Debugging?

nph4rd replied:
Indeed :) Thank you!

mucinoab reviewed (commented):
LGTM but I think it would be more ergonomic to have the `OptionalOrderParams` and such wrapped in an `Option` (`optional_params: Option<OptionalParams<'_>>`)  because if you don't need any optional parameters you just have to pass `None` instead of having to construct the whole structure.

nph4rd committed:
Fix link

nph4rd committed:
Get rid of println in client.rs

arturomf94 commented:
> LGTM but I think it would be more ergonomic to have the `OptionalOrderParams` and such wrapped in an `Option` (`optional_params: Option<OptionalParams<'_>>`) because if you don't need any optional parameters you just have to pass `None` instead of having to construct the whole structure.

This is a really good point! I'll make this change before merging.

nph4rd committed:
Wrap OptionalParams and OptionalOrderParams in Option

This avoids constructing the object when there are no
optional parameters involved in the call.

nph4rd committed:
Constuct maps when there are optional parameters in body

arturomf94 commented:
@mucinoab - I've made the changes you suggested. I also included a more explicit construction of the parameter `Map` when there are optional parameters included so as not to include any empty parameters in the body. Let me know what you think.
